src/AppInstallerCommonCore/Fonts.cpp

```diff
@@ -44,20 +67,647 @@ namespace AppInstaller::Fonts
+                    // resolve the path correctly.
+                    fileInfo.Title = GetFontFileTitle(fileInfo.FilePath);
+                }
+                CATCH_LOG();
+            }
+        }
+        else
+        {
+            fileInfo.WinGetSupported = false;
+            fileInfo.FileType = DWRITE_FONT_FILE_TYPE_UNKNOWN;
+        }
+
+        if (fileInfo.Title.empty())
+        {
+            // If the title is still empty (such as failure to set it from the File), use the Filename
+            fileInfo.Title = fileInfo.FilePath.filename().wstring();
+        }
+
+        // This is the expected full registry path of the font file.
+        fileInfo.RegistryPath = GetFontRegistryPath(context);
+
+        // This is the expected full install path of the font file, were it to be installed.
+        if (fileInfo.InstallerSource != InstallerSource::UWP)
+        {
+            fileInfo.InstallPath = GetFontFileInstallPath(context) / fileInfo.FilePath.filename();
+        }
+        else
+        {
+            // For UWP it is the UWP package that decides where the font is to be, which we cannot
+            // know and wouldn't ever be trying to install, so this scenario must be a query for
+            // information. If it is a query, then the install path is the file path.
+            fileInfo.InstallPath = fileInfo.FilePath;
+        }
+
+        // Check whether the install path is in the registry.
+        const auto& registryTitle = CheckRegistryForFontFileReference(fileInfo.RegistryPath.value(), fileInfo.InstallPath, fileInfo.Scope);
+
+        // Use the registry title as the title if we still dont' know.
+        if (fileInfo.Title.empty())
+        {
+            if (registryTitle.has_value())
+            {
+                fileInfo.Title = registryTitle.value();
+            }
+            else
+            {
+                // If the title is still empty, use the Filename
+                fileInfo.Title = fileInfo.FilePath.filename().wstring();
+            }
+        }
+
+        // Confirm install states
+        fileInfo.IsFontFileInstalled = std::filesystem::exists(fileInfo.InstallPath) ? true : false;
+        fileInfo.IsFontFileRegistered = registryTitle.has_value() ? true : false;
+
+        // Set status
+        if (fileInfo.IsFontFileInstalled && fileInfo.IsFontFileRegistered)
+        {
+            fileInfo.Status = FontStatus::OK;
+        }
+        else if (fileInfo.IsFontFileInstalled || fileInfo.IsFontFileRegistered)
+        {
+            fileInfo.Status = FontStatus::Corrupt;
+        }
+        else
+        {
+            fileInfo.Status = FontStatus::Absent;
+        }
+
+        return fileInfo;
+    }
+
+    FontValidationResult ValidateFontPackage(FontContext& context)
+    {
+        auto result = FontValidationResult();
+
+        AICLI_LOG(Core, Info, << L"Validating font package: " << context.PackageIdentifier.value_or(L"Unknown").c_str());
```

Comment:
```suggestion
        AICLI_LOG(Core, Info, << L"Validating font package: " << context.PackageIdentifier.value_or(L"<Unknown>").c_str());
```
Putting `<>` catches the eye more to me, making the default value stand out more (and differentiating it from the potentially valid value of "Unknown").